service/src/main/java/org/cbioportal/service/impl/CopyNumberCountServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package org.cbioportal.service.impl;
+
+import org.cbioportal.model.CopyNumberCount;
+import org.cbioportal.model.CopyNumberSampleCountByGene;
+import org.cbioportal.model.GeneticProfile;
+import org.cbioportal.service.CopyNumberCountService;
+import org.cbioportal.service.DiscreteCopyNumberService;
+import org.cbioportal.service.GeneticDataService;
+import org.cbioportal.service.GeneticProfileService;
+import org.cbioportal.service.exception.GeneticProfileNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CopyNumberCountServiceImpl implements CopyNumberCountService {
+
+    @Autowired
+    private GeneticDataService geneticDataService;
+    @Autowired
+    private DiscreteCopyNumberService discreteCopyNumberService;
+    @Autowired
+    private GeneticProfileService geneticProfileService;
+
+    @Override
+    public List<CopyNumberCount> fetchCopyNumberCounts(String geneticProfileId, List<Integer> entrezGeneIds,
+                                                       List<Integer> alterations)
+        throws GeneticProfileNotFoundException {
+
+        validateGeneticProfile(geneticProfileId);
+
+        Integer numberOfSamplesInGeneticProfile = geneticDataService.getNumberOfSamplesInGeneticProfile(
+            geneticProfileId);
+        List<CopyNumberSampleCountByGene> copyNumberSampleCountByGeneList = discreteCopyNumberService
+            .getSampleCountByGeneAndAlteration(geneticProfileId, entrezGeneIds, alterations);
+
+        List<CopyNumberCount> copyNumberCounts = new ArrayList<>();
+        for (int i = 0; i < alterations.size(); i++) {
+            Integer alteration = alterations.get(i);
+            Integer entrezGeneId = entrezGeneIds.get(i);
```

Comment:
@ersinciftci already merged but, seems like we are assuming here that `alterations.size() == entrezGeneIds.size()`